### PR TITLE
Enhance rewriting rules

### PR DESCRIPTION
**Pull request by lexmag**

### CODE REVIEW

File: lib/elixir/src/elixir_rewrite.erl [hunk 1]
```diff
@@ -184,15 +185,19 @@ rewrite(?access, _DotMeta, 'get', Meta, [Arg, _], Env)
   elixir_errors:compile_error(Meta, ?m(Env, file),
     "the Access syntax and calls to Access.get/2 are not available for the value: ~ts",
     ['Elixir.Macro':to_string(Arg)]);
-rewrite(?string_chars, _DotMeta, 'to_string', _Meta, [String], _File) when is_binary(String) ->
+rewrite(?list_chars, _DotMeta, 'to_char_list', _Meta, [List], _Env) when is_list(List) ->
```
Comment (josevalim) [inline lib/elixir/src/elixir_rewrite.erl]: This is not right. We need to check that all elements in the list are integers and that they represent codepoints.

Comment (lexmag) [inline lib/elixir/src/elixir_rewrite.erl]: Alright, I thought, for a start we could just go same way as we do in `List.Chars` for lists: https://github.com/elixir-lang/elixir/blob/master/lib/elixir/lib/list/chars.ex#L36.
Shall we restrict there as well?

Comment (lexmag) [inline lib/elixir/src/elixir_rewrite.erl]: Although might fine for performance reasons there.

Comment (josevalim) [inline lib/elixir/src/elixir_rewrite.erl]: You are right. If we are not checking there, it doesn't make sense to check here. You may want to add a note above those implementations though to let the developer reading the code know they are being inlined :D

**Pull request merged**